File: Java_MVVM/app/src/main/java/eu/gotoinc/requesinjava_mvvm/adapter/ViewHolder.java
package eu.gotoinc.requesinjava_mvvm.adapter;

import androidx.databinding.BaseObservable;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;
import eu.gotoinc.requesinjava_mvvm.BR;

class ViewHolder extends RecyclerView.ViewHolder {
	private final ViewDataBinding binding;

	ViewHolder(ViewDataBinding binding) {
		super(binding.getRoot());
		this.binding = binding;
	}


	void bind(BaseObservable baseViewModel) {
		binding.setVariable(BR.vm, baseViewModel);
		binding.executePendingBindings();
	}

}

File: Kotlin_MVVM/app/src/main/java/eu/gotoinc/kotlinmvvm/adapter/ViewHolder.kt
package eu.gotoinc.kotlinmvvm.adapter

import androidx.databinding.BaseObservable
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView
import eu.gotoinc.kotlinmvvm.BR

class ViewHolder(private val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {

    fun bind(baseViewModel: BaseObservable) {
        binding.apply {
            setVariable(BR.viewModel, baseViewModel)
            executePendingBindings()
        }
    }

}